packages/ember-glimmer/lib/utils/references.js

```diff
@@ -120,7 +120,14 @@ export class GetHelperReference extends CachedReference {
   isDirty() { return true; }
 
   value() {
-    return this.sourceReference.get(this.pathReference.value()).value();
+    let key = this.pathReference.value();
+    let keyType = typeof key;
+
+    if (key && (keyType === 'string' || keyType === 'number')) {
+      return this.sourceReference.get(key).value();
+    }
+
+    return NULL_REFERENCE.value();
```

Comment:
I didn't realize we are at the "leaf" of a chain here. In that case, we can just return `null` here. `NULL_REFERENCE` is just a POJO whose `value` and `get` is hardcoded to always return `null`.